src/main/java/org/jenkinsci/plugins/junitrealtimetestreporter/PipelineRealtimeTestResultAction.java

```diff
@@ -42,13 +52,21 @@
     private final String workspace;
     private final boolean keepLongStdio;
     private final String glob;
+    private final StepContext context;
```

Comment:
Just make sure this is not serialized to `build.xml`!

Reply:
Does not appear to be:

```
    <org.jenkinsci.plugins.junitrealtimetestreporter.PipelineRealtimeTestResultAction plugin="junit-realtime-test-reporter@999999-SNAPSHOT">
      <descriptions class="concurrent-hash-map"/>
      <id>14</id>
      <node></node>
      <workspace>/Users/timja/code/jenkins/junit-realtime-test-reporter-plugin/work/workspace/junit-attachments-test_PR-8@2</workspace>
      <keepLongStdio>false</keepLongStdio>
      <glob>**/target/surefire-reports/TEST-*.xml</glob>
    </org.jenkinsci.plugins.junitrealtimetestreporter.PipelineRealtimeTestResultAction>
```

Reply:
Hmm, but `PipelineRealtimeTestResultAction` is? Then this looks dangerous. Suggest making it `transient` and studying the conditions under which it might be saved.

Reply:
Hmm, but `PipelineRealtimeTestResultAction` is? Then this looks dangerous. Suggest making it `transient` and studying the conditions under which it might be saved.

Reply:
Javadoc says it's safe to save it? https://javadoc.jenkins.io/plugin/workflow-step-api/org/jenkinsci/plugins/workflow/steps/StepContext.html

Reply:
Safe to serialize it to `program.dat`! Not via XStream.

Reply:
Managed to save it, only has this in it:

```xml
      <context class="org.jenkinsci.plugins.workflow.cps.CpsStepContext" plugin="workflow-cps@2648.va9433432b33c">
        <executionRef class="flow-owner" plugin="workflow-job@1145.v7f2433caa07f">
          <job>junit-attachments-test/PR-8</job>
          <id>17</id>
        </executionRef>
        <id>14</id>
        <bodyHeads>
          <int>1</int>
        </bodyHeads>
        <body class="org.jenkinsci.plugins.workflow.cps.HandleBodyReference">
          <id>9</id>
        </body>
        <threadId>8</threadId>
        <stepDescriptorId>org.jenkinsci.plugins.junitrealtimetestreporter.RealtimeJUnitStep</stepDescriptorId>
      </context>

```
Looks fine?

Reply:
I…guess. Would be safer to not do this.

There is no new test coverage. What caused the original error? `RealtimeJUnitStepTest.smokes` already tested controller restarts.

Reply:
Maybe only when `allowEmptyResults` is in use?

Reply:
Empty results, the Junit plugin assumes it gets a non null task listener and it logs something when there's no results I think (although probably isn't annotating the methods as it's not really expected to have these methods called outside of the plugin).

Reply:
I'll take a look tomorrow, how do I get to the `FlowExecutionOwner`?

Reply:
checkcast `Run` to `FEO.Executable`

Reply:
See https://github.com/jenkinsci/junit-realtime-test-reporter-plugin/pull/27